### DISCUSSION TITLE
Tutorials or Blog posts that should exist

### DISCUSSION THREAD

**Discussion by hayes** [hayes/pothos#109] [Ideas]
I want to create a series of blog posts that are tutorials for GiraphQL

* GiraphQL + next.js
* next.js + scope auth
* GiraphQL + graphql helix + deploy somewhere
* Full stack type safety (react, giraphql, prisma)
* Migrating from SDL to code first
* giraphql + next.js + relay

**Comment by smolattack** [comment]
A basic GiraphQL + next.js and scope auth example would be great! Any chance it can be implemented using Relay?

**Reply by hayes** [reply]
This is mostly aspirational.  Writing is not my string suite, so in reality, I'm not sure when or if any of these will get written.  Maybe someone else will end up writing something at some point.

**Reply by smolattack** [reply]
In that case, may I suggest adding an `examples` folder to this repo and just upload one or more working nextjs apps? Maybe sprinkle a few comments here and there. Even without an accompanying blog, a working example could really help someone determined to experiment with GiraphQL. And if you do decide to write an article, at least you will already have an app to to write about.

I think the docs are pretty well written so don't put yourself down!

**Reply by NICHTJ3** [reply]
I would say relay and nextjs are a little strange to work with (mostly the SSR stuff not being directly supported and suspense in SSR) and deployments for a next app with a graphql endpoint can be finicky due to serverless functions

From my experience I would say you're pretty much always better off starting with a spa unless you need SEO e.g. A social media site.

**Reply by smolattack** [reply]
That's good to know (SSR not officially supported). Still a basic example would be nice (either nextjs or SPA)

**Reply by NICHTJ3** [reply]
Sorry @smolattack  I just want to clarify it's not that they don't officially support SSR 🤦‍♂️ It's that they don't promote a pattern around it and the patterns that have been shown in the relay issues frequently have edge cases or perform needless fetches causing more network overhead than necessary

**Reply by smolattack** [reply]
@NICHTJ3, I see what you mean. Yes, I've seen issues where fetches can happen on both the frontend and backend (essentially one being wasted). I think there are workarounds where you can detect the environment you're in and determine if the fetch should be made or not but I'm no expert and have no idea how that works.

**Reply by NICHTJ3** [reply]
Yeah I've just created empty environments on the server to work around this but it would be nice to have a well defined pattern
